corehq/warehouse/models/shared.py

```diff
@@ -0,0 +1,86 @@
+import os
+
+from django.db import connections
+from django.conf import settings
+from django.template import Context, engines
+
+from corehq.sql_db.routers import db_for_read_write
+
+from corehq.warehouse.transforms import get_transform
+from corehq.warehouse.const import (
+    DJANGO_MAX_BATCH_SIZE,
+)
+
+
+class WarehouseTableMixin(object):
+
+    @classmethod
+    def sql_query_template_name(cls):
+        return None
+
+    @classmethod
+    def dependencies(cls):
+        raise NotImplementedError
+
+    @classmethod
+    def load(cls):
```

Comment:
i haven't used upsert before but it looks like it would solve this problem without the need for another intermediate table